### ISSUE TITLE
Feature Request: Realistic tool crafting/smelting/blacksmithing

### ISSUE THREAD

**Issue by DonFlymoor** [minetest-LOTR/Lord-of-the-Test#175]
The current LOTR crafting system is decidedly fake. In the books, weapons were forged - not crafted by sticking some metal together. I propose a more LOTR -themed crafting system. Some items should be craftable by hand, like bows and stone items, but metal tools and other items uncraftable by hand should be created by heating ingots in a forge, then forging them with a hammer on an anvil. In addition, crafting should not be instantaneous. It should take in-game minutes (1 min ~=1 second) to hours (60 seconds) to create items, depending on the complexity. This will also make items feel more valuable, since you didn't get them by just putting resources into a grid, and force you to prepare yourself before heading into a fight.

**Comment by lumidify** [comment]
Hi,

I had very similar ideas a while ago, so I actually started implementing a new crafting system (in Minetest: Third Age), but I didn't have any time to continue working on it. The current work is in `lottcrafting/` and includes a crafting system that allows adding recipes with custom attributes so the recipes for different types of furnaces, etc. can be separated. It's all a bit WIP, though, and it is currently only used by the crafting tables, the forge, and the smelting furnace (none of which actually have any crafting recipes defined yet). At one point, I also wanted to add a spinning wheel and loom, but I have no idea what happened to those. If you (or anyone else) want to work on this, you're more than welcome to. I can try to help with using the crafting system and deciding what would be good to implement, but I sadly don't have enough time to do anything big myself. If you're interested, I can also give you my logs from when this was discussed a long time ago in our IRC channel (it would be best if you asked me for that on IRC).

**Comment by Amaz1** [comment]
Yes, this is something that really needs work doing - in fact, I would say it, along with a good tool system, is the next main thing that needs to be tackled with MTTA... I've attempted various things, but never really come up with a satisfactory result. Here's the progress I made last time with tools: [lotttools.zip](https://github.com/minetest-LOTR/Lord-of-the-Test/files/6328078/lotttools.zip) It's not much, but it's a start... If you want to have a look at that, and come up with something, in conjunction with lottcrafting, as lumidify pointed out, that would be great.
I like the idea of taking actual in-game time for the crafting.

**Comment by DonFlymoor** [comment]
We could also integrate #151 into the crafting system, you starting out with no knowledge on how to craft things.
